madsim/src/sim/runtime/mod.rs

```diff
@@ -226,6 +231,40 @@ impl Handle {
         context::current(|h| h.clone())
     }
 
+    /// Returns a [`Handle`] view over the currently running [`Runtime`]
+    ///
+    /// Returns an error if no Runtime has been started
+    ///
+    /// Contrary to `current`, this never panics
+    pub fn try_current() -> Result<Self, TryCurrentError> {
+        context::try_current(|h| h.clone()).ok_or(TryCurrentError)
+    }
+
+    /// spawn a task
+    pub fn spawn<F>(&self, future: F) -> JoinHandle<F::Output>
+    where
+        F: Future + Send + 'static,
+        F::Output: Send + 'static,
+    {
+        self.create_node()
+            .name("spawn a task")
```

Comment:
We shouldn't create a new node here. The task should be spawned within the current node. Please use `Spawner::current()` instead.

Reply:
I have tried `Spawner::current` before, but in some cases, like dropping some `Future` and calling a sync `drop` to spawn an async task which means there are no tasks in the runtime. The drop site is out the scope of https://github.com/madsim-rs/madsim/blob/ca933529dca42bcc5203a98268365fd778a05676/madsim/src/sim/task/mod.rs#L270

Reply:
I see. In this case, I think we should enter task context before dropping the task. And here we should still use `Spawner::current()`.